src/common/com/intellij/plugins/haxe/ide/HaxeControllingCompletionContributor.java

```diff
@@ -111,8 +111,8 @@ private static boolean shouldRemoveDuplicateCompletions(PsiFile file) {
   private static Set<CompletionResult> removeDuplicateCompletions(Set<CompletionResult> unfilteredCompletions) {
     // We sort the elements according to name, giving preference to compiler-provided results
     // if the two names are identical.
```

Comment:
Here, we weren't removing duplicate completions properly because I ended up sorting a temporary array (sorted.toArray() at 115).  I rewrote this so it works with Java6 and 8.